### PR TITLE
Ensure server is providing full cert chain

### PR DESCRIPTION
**Pull request by externl**
When using a certificate where the intermediate cert is in the system store,  SslStream will include it by default, but Quic will not.

This PR updates the hello server to ensure the full certificate chain is always provided. We do this by explicitly setting a `SslStreamCertificateContext`. For simplicity, I've split the `server.p12` into a separate cert and key. 
This is what's provided by let's encrypt, is standard when getting a cert from other CAs, and it's easier to maintain one code path.

### CODE REVIEW

File: src/Hello/Hello.cs [hunk 1]
```diff
@@ -9,16 +9,27 @@
 // The log level for the logger factory
 LogLevel logLevel = Enum.Parse<LogLevel>(Environment.GetEnvironmentVariable("LOG_LEVEL") ?? "Debug");
 
-// The path to the server certificate used for TLS.
-string serverCert = Environment.GetEnvironmentVariable("SERVER_CERT") ?? "/certs/server.p12";
+// The private key for the server certificate.
+string serverKey = Environment.GetEnvironmentVariable("SERVER_KEY") ?? "/certs/privkey.pem";
+
+// The server certificate (with the full chain) file.
+string serverCert = Environment.GetEnvironmentVariable("SERVER_CERT") ?? "/certs/fullchain.pem";
```
Comment (pepone) [inline src/Hello/Hello.cs:16]: and `/certs/server_cert.pem`

File: src/Hello/Hello.cs [hunk 2]
```diff
@@ -9,16 +9,27 @@
 // The log level for the logger factory
 LogLevel logLevel = Enum.Parse<LogLevel>(Environment.GetEnvironmentVariable("LOG_LEVEL") ?? "Debug");
 
-// The path to the server certificate used for TLS.
-string serverCert = Environment.GetEnvironmentVariable("SERVER_CERT") ?? "/certs/server.p12";
+// The private key for the server certificate.
+string serverKey = Environment.GetEnvironmentVariable("SERVER_KEY") ?? "/certs/privkey.pem";
```
[the remark this answers is not in the document]
Comment (externl) [reply src/Hello/Hello.cs:13]: `privkey.pem` is what's used by let's encrypt, I'd rather keep it

File: src/Hello/Hello.cs [hunk 3]
```diff
@@ -9,16 +9,27 @@
...
+using var serverCertificate = X509Certificate2.CreateFromPemFile(serverCert, serverKey);
+
+// Create a collection with the server certificate and any intermediate certificates.
+var intermediates = new X509Certificate2Collection();
+intermediates.ImportFromPemFile(serverCert);
+
 // Create server authentication options with the server certificate.
 var sslAuthenticationOptions = new SslServerAuthenticationOptions
 {
     ServerCertificate = new X509Certificate2(serverCert),
```
Comment (pepone) [inline src/Hello/Hello.cs:31]: Don't need to set `ServerCertificate` in this case.

**Review by pepone — APPROVED** [review]

File: src/Hello/Hello.cs (hunk 1 above)
Comment (externl) [reply src/Hello/Hello.cs:16]: `fullchain.pem` is what's used by let's encrypt. I'd prefer to keep it.

Comment (bernardnormier) [reply src/Hello/Hello.cs:16]: What about changing the env variable names then?

**Comment by externl** [comment]
Maybe we should just require the user specify a path to both and complain if it's missing. that way we don't have to worry about it and our compose file will set the local ones accordingly,. 

The names aren't great since there's the fullchian.pem is not a chain.

**Comment by externl** [comment]
I renamed the files and the env vars

File: .vscode/cspell.json [hunk 4]
```diff
@@ -4,10 +4,12 @@
     "words": [
         "buildtransitive",
         "contentfiles",
+        "fullchain",
+        "privkey",
```
Comment (pepone) [inline .vscode/cspell.json:8]: I guess you can remove those now.

**Pull request merged**